lib/azure/armrest/virtual_machine_service.rb

```diff
@@ -71,23 +71,26 @@ def generalize(vmname, group = configuration.resource_group)
       # Retrieves the settings of the VM named +vmname+ in resource group
       # +group+, which will default to the same as the name of the VM.
       #
-      # By default this method will retrieve the model view. If the +model_view+
-      # parameter is false, it will retrieve an instance view. The difference is
-      # in the details of the information retrieved.
+      # You can also specify any query options. At this time only the
+      # :expand => 'instanceView' option is supported, but others could
+      # be added over time.
       #
-      def get(vmname, group = configuration.resource_group, model_view = true)
-        model_view ? super(vmname, group) : get_instance_view(vmname, group)
+      def get(vmname, group = configuration.resource_group, options = {})
```

Comment:
the argument list still changed, which breaks the compatibility.

Reply:
Eh, it doesn't break existing behavior, though, since the 3rd argument defaults to empty.

Reply:
It breaks. consider a call with `get(vm, group, false)`. The rest call may fail, or least not returning what it used to.

Reply:
Ah, true. I'd rather break the api than add another method, though, because this is more flexible going forward.

Reply:
With a minor effort you can at least check whether `options` is a `Hash`. If not, treat it as a boolean.

Reply:
@bzwei Ok, updated.